src/Nancy/RequestHeaders.cs

```diff
@@ -372,17 +428,19 @@ private static IEnumerable<INancyCookie> GetNancyCookies(IEnumerable<string> coo
 
         private IEnumerable<string> GetValue(string name)
         {
-            return this.GetValue(name, x => x);
+            return this.GetValue(name, x => x, new string[] {});

```

Comment:
For `Enumerable.Empty` you'd have to allocate an enumerator for each time as well, so that makes sense. I think we should at least cache this instance, if we don't want a generic empty array cache.